vpc/service/branch_enis.go

```diff
@@ -1404,3 +1406,64 @@ func (vpcService *vpcService) createBranchENI(ctx context.Context, tx *sql.Tx, s
 
 	return iface, nil
 }
+
+func (vpcService *vpcService) ListTrunkToBranchENIMappings(ctx context.Context, req *vpcapi.ListTrunkToBranchENIMappingsRequest) (*vpcapi.ListTrunkToBranchENIMappingsResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "listTrunkToBranchENIMappings")
+	defer span.End()
+
+	mappings := []*vpcapi.TrunkENIMapping{}
+
+	tx, err := vpcService.db.BeginTx(ctx, &sql.TxOptions{
+		ReadOnly:  true,
+		Isolation: sql.LevelRepeatableRead,
+	})
+	if err != nil {
+		err = errors.Wrap(err, "Cannot start database transaction")
+		tracehelpers.SetStatus(err, span)
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			tx.Rollback()
+		}
+		err = tx.Commit()
+	}()
+
+	rows, err := tx.QueryContext(ctx, `
+	SELECT
+	branch_eni,
+	trunk_eni
+	FROM branch_eni_attachments WHERE state = 'attached'
+`)
+	defer rows.Close()
```

Comment:
This should be after the err check

Reply:
Why? Wouldn't we want to always close the rows object even on error? or is the concern that the close method will crash somehow if the rows object isn't legit in the first place?

Reply:
QueryContext return (*Row, error). Given this, there's no guarantee that Row will be non-nil. See code:
https://github.com/lib/pq/blob/f3b22b2cad9e4567803b7ffd9853b8acda021438/conn.go#L830-L832

Often times when you have (interface{}, error) or (pointer,error) in golang, the pointer / interface is nil'd fi error != nil

Reply:
got it.